tests/test_bpscore.py

```diff
@@ -0,0 +1,94 @@
+import os
+import shutil
+import tempfile
+import unittest
+from random import choice
+from string import ascii_letters
+import lsst.ctrl.bps.bps_core as bps
+
+
+class TestExecute(unittest.TestCase):
+
+    def setUp(self):
+        self.file = tempfile.NamedTemporaryFile("w+")
+
+    def tearDown(self):
+        self.file.close()
+
+    def testSuccessfulExecution(self):
+        """Test exit status if command succeeded."""
+        status = bps.execute('true', self.file.name)
+        self.assertIn('true', self.file.read())
+        self.assertEqual(status, 0)
+
+    def testFailingExecution(self):
+        """Test exit status if command failed."""
+        status = bps.execute('false', self.file.name)
+        self.assertIn('false', self.file.read())
+        self.assertNotEqual(status, 0)
+
+
+class TestPrettyPrinting(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def testTaskLabelFormatting(self):
+        """Test if a task label is properly formatted."""
+        original = "'raw+{band: i, instrument: HSC, detector: 17}'"
+        expected = "'raw\nband=i\n instrument=HSC\n detector=17'"
+        result = bps.pretty_dataset_label(original)
+        self.assertEqual(result, expected)
+
+
+class TestBpsCore(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def testCreatingQuantumGraph(self):
+        """Test if an appropriately named pickle file created."""
+        directory = tempfile.mkdtemp(prefix="bps")
+        filename = "".join(choice(ascii_letters) for _ in range(8))
+        path = os.path.join(directory, f"{filename}.pickle")
+        settings = {
+            "global": {
+                "createQuantumGraph": "touch {qgraphfile}",
+                "submitPath": directory,
+                "uniqProcName": filename,
```

Comment:
Does this test really need uniqProcName?

Reply:
To my current understanding `uniqProcName` defines the name of the file which normally holds the quantum graph that is line 243 in `bps_core.py` reads:
```python
self.qgraph_filename = "%s/%s.pickle" % (self.submit_path, self.config["uniqProcName"])
```
If not set, `self.config["uniqProcName"]` defaults to an empty string. So I'd say it's not strictly required; after all `.pickle` is a valid filename. I just thought it will be less confusing for the pickle file to have an actual name. But I see no problems with removing it.

Reply:
To my current understanding `uniqProcName` defines the name of the file which normally holds the quantum graph that is line 243 in `bps_core.py` reads:
```python
self.qgraph_filename = "%s/%s.pickle" % (self.submit_path, self.config["uniqProcName"])
```
If not set, `self.config["uniqProcName"]` defaults to an empty string. So I'd say it's not strictly required; after all `.pickle` is a valid filename. I just thought it will be less confusing for the pickle file to have an actual name. But I see no problems with removing it.